source/extensions/filters/http/json_to_metadata/filter.h

```diff
@@ -94,37 +101,52 @@ class Filter : public Http::PassThroughFilter, Logger::Loggable<Logger::Id::filt
                                           bool end_stream) override;
   Http::FilterDataStatus decodeData(Buffer::Instance& data, bool end_stream) override;
   Http::FilterTrailersStatus decodeTrailers(Http::RequestTrailerMap&) override;
+  Http::FilterHeadersStatus encodeHeaders(Http::ResponseHeaderMap& headers,
+                                          bool end_stream) override;
+  Http::FilterDataStatus encodeData(Buffer::Instance& data, bool end_stream) override;
+  Http::FilterTrailersStatus encodeTrailers(Http::ResponseTrailerMap&) override;
 
 private:
   using StructMap = absl::flat_hash_map<std::string, ProtobufWkt::Struct>;
   // Handle on_missing case of the `rule` and store in `struct_map`.
-  void handleOnMissing(const Rule& rule, StructMap& struct_map);
+  void handleOnMissing(const Rule& rule, StructMap& struct_map,
+                       Http::StreamFilterCallbacks& filter_callback);
   // Handle on_present case of the `rule` and store in `struct_map`, which depends on
   // the value of `parent_node->key`.
   absl::Status handleOnPresent(Json::ObjectSharedPtr parent_node, const std::string& key,
-                               const Rule& rule, StructMap& struct_map);
+                               const Rule& rule, StructMap& struct_map,
+                               Http::StreamFilterCallbacks& filter_callback);
 
   // Process the case without body, i.e., on_missing is applied for all rules.
-  void handleAllOnMissing(const Rules& rules, bool& processing_finished_flag);
+  void handleAllOnMissing(const Rules& rules, bool& processing_finished_flag,
+                          Http::StreamFilterCallbacks& filter_callback, bool shouldClearRouteCache);
```

Comment:
snake style in *.h file

also input variable should in front of output variable. `processing_finished_flag` and `filter_callback` are probably need to be after should_clear_route_cache
https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs